test/e2e_node/container_manager_test.go

```diff
@@ -92,50 +92,65 @@ var _ = framework.KubeDescribe("Kubelet Container Manager [Serial]", func() {
-				var err error
-				podClient := f.PodClient()
-				podName := "besteffort" + string(uuid.NewUUID())
-				podClient.Create(&v1.Pod{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: podName,
-					},
-					Spec: v1.PodSpec{
-						Containers: []v1.Container{
-							{
-								Image: "gcr.io/google_containers/serve_hostname:v1.4",
-								Name:  podName,
+			Context("", func() {
+				It("pod infra containers oom-score-adj should be -998 and best effort container's should be 1000", func() {
+					var err error
+					podClient := f.PodClient()
+					podName := "besteffort" + string(uuid.NewUUID())
+					podClient.Create(&v1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: podName,
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Image: "gcr.io/google_containers/serve_hostname:v1.4",
+									Name:  podName,
+								},
 							},
 						},
-					},
-				})
-				var pausePids []int
-				By("checking infra container's oom-score-adj")
-				Eventually(func() error {
-					pausePids, err = getPidsForProcess("pause", "")
-					if err != nil {
-						return fmt.Errorf("failed to get list of pause pids: %v", err)
-					}
-					for _, pid := range pausePids {
-						if err := validateOOMScoreAdjSetting(pid, -998); err != nil {
-							return err
+					})
+					var pausePids []int
+					By("checking infra container's oom-score-adj")
+					Eventually(func() error {
+						pausePids, err = getPidsForProcess("pause", "")
+						if err != nil {
+							return fmt.Errorf("failed to get list of pause pids: %v", err)
 						}
+						for _, pid := range pausePids {
+							if err := validateOOMScoreAdjSetting(pid, -998); err != nil {
+								return err
+							}
+						}
+						return nil
+					}, 2*time.Minute, time.Second*4).Should(BeNil())
+					var shPids []int
+					By("checking besteffort container's oom-score-adj")
+					Eventually(func() error {
+						shPids, err = getPidsForProcess("serve_hostname", "")
+						if err != nil {
+							return fmt.Errorf("failed to get list of serve hostname process pids: %v", err)
+						}
+						if len(shPids) != 1 {
+							return fmt.Errorf("expected only one serve_hostname process; found %d", len(shPids))
+						}
+						return validateOOMScoreAdjSetting(shPids[0], 1000)
+					}, 2*time.Minute, time.Second*4).Should(BeNil())
+				})
+				// Log the running containers and pods here to help debug #37580.
+				// TODO(random-liu): Remove this after the issue is resolved.
+				AfterEach(func() {
+					if CurrentGinkgoTestDescription().Failed {
+						By("Dump all running docker containers")
+						output, err := exec.Command("docker", "ps").CombinedOutput()
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("Running docker containers: %q", string(output))
+						By("Dump all running pods")
+						output, err = exec.Command("curl", "-s", "http://localhost:10255/pods").CombinedOutput()
```

Comment:
I think we have utilities in the e2e framework to do this through the apiserver. Can we reuse that? 

If not, will this output be formatted/readable?